### ISSUE TITLE
Xabber 0.10.55 crashes for various reasons

### ISSUE THREAD

**Issue by aelmahmoudy** [redsolution/xabber-android#392]
When I attempted to add a new contact, Xabber crashed, here's the logcat:

```
I/com.xabber.android.ui.ContactAdd@422c1ed0(28059): onPause
W/System.err(28059): com.xabber.android.data.Application@41b5f6e0
I/System.out(28059): 07:39:25 PM SENT (1108581952): <presence id="T4MFB-28" to="new.contact@jabber.ru" type="subscribe"><c xmlns='http://jabber.org/protocol/caps' hash='sha-1' node='http://www.igniterealtime.org/projects/smack/' ver='2SZTp6+uM08fNDqEoW/+dmBqrQ4='/></presence>
W/System.err(28059): com.xabber.android.data.NetworkException
W/System.err(28059): 	at com.xabber.android.data.roster.RosterManager.setGroups(RosterManager.java:332)
W/System.err(28059): 	at com.xabber.android.ui.GroupEditorFragment.onPause(GroupEditorFragment.java:158)
W/System.err(28059): 	at android.support.v4.app.Fragment.performPause(Fragment.java:1947)
W/System.err(28059): 	at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1003)
W/System.err(28059): 	at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1136)
W/System.err(28059): 	at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1118)
W/System.err(28059): 	at android.support.v4.app.FragmentManagerImpl.dispatchPause(FragmentManager.java:1941)
W/System.err(28059): 	at android.support.v4.app.FragmentActivity.onPause(FragmentActivity.java:390)
W/System.err(28059): 	at com.xabber.android.ui.helper.ManagedActivity.onPause(ManagedActivity.java:47)
W/System.err(28059): 	at android.app.Activity.performPause(Activity.java:5387)
W/System.err(28059): 	at android.app.Instrumentation.callActivityOnPause(Instrumentation.java:1233)
W/System.err(28059): 	at android.app.ActivityThread.performPauseActivity(ActivityThread.java:3034)
W/System.err(28059): 	at android.app.ActivityThread.performPauseActivity(ActivityThread.java:3003)
W/System.err(28059): 	at android.app.ActivityThread.handlePauseActivity(ActivityThread.java:2981)
W/System.err(28059): 	at android.app.ActivityThread.access$1000(ActivityThread.java:135)
W/System.err(28059): 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1213)
W/System.err(28059): 	at android.os.Handler.dispatchMessage(Handler.java:102)
W/System.err(28059): 	at android.os.Looper.loop(Looper.java:136)
W/System.err(28059): 	at android.app.ActivityThread.main(ActivityThread.java:5001)
W/System.err(28059): 	at java.lang.reflect.Method.invokeNative(Native Method)
W/System.err(28059): 	at java.lang.reflect.Method.invoke(Method.java:515)
W/System.err(28059): 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:785)
W/System.err(28059): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:601)
W/System.err(28059): 	at dalvik.system.NativeStart.main(Native Method)
I/com.xabber.android.ui.ContactList@41ca4a40(28059): onResume
D/AndroidRuntime(28059): Shutting down VM
W/dalvikvm(28059): threadid=1: thread exiting with uncaught exception (group=0x4184dba8)
E/AndroidRuntime(28059): FATAL EXCEPTION: main
E/AndroidRuntime(28059): Process: com.xabber.androiddev, PID: 28059
E/AndroidRuntime(28059): java.lang.NullPointerException
E/AndroidRuntime(28059): 	at com.xabber.android.ui.adapter.GroupedContactAdapter.getView(GroupedContactAdapter.java:182)
E/AndroidRuntime(28059): 	at android.widget.AbsListView.obtainView(AbsListView.java:2257)
E/AndroidRuntime(28059): 	at android.widget.ListView.makeAndAddView(ListView.java:1790)
E/AndroidRuntime(28059): 	at android.widget.ListView.fillSpecific(ListView.java:1337)
E/AndroidRuntime(28059): 	at android.widget.ListView.layoutChildren(ListView.java:1608)
E/AndroidRuntime(28059): 	at android.widget.AbsListView.onLayout(AbsListView.java:2089)
E/AndroidRuntime(28059): 	at android.view.View.layout(View.java:14824)
E/AndroidRuntime(28059): 	at android.view.ViewGroup.layout(ViewGroup.java:4631)
E/AndroidRuntime(28059): 	at android.widget.RelativeLayout.onLayout(RelativeLayout.java:1055)
E/AndroidRuntime(28059): 	at android.view.View.layout(View.java:14824)
E/AndroidRuntime(28059): 	at android.view.ViewGroup.layout(ViewGroup.java:4631)
E/AndroidRuntime(28059): 	at android.widget.RelativeLayout.onLayout(RelativeLayout.java:1055)
E/AndroidRuntime(28059): 	at android.view.View.layout(View.java:14824)
E/AndroidRuntime(28059): 	at android.view.ViewGroup.layout(ViewGroup.java:4631)
E/AndroidRuntime(28059): 	at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1671)
E/AndroidRuntime(28059): 	at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1525)
E/AndroidRuntime(28059): 	at android.widget.LinearLayout.onLayout(LinearLayout.java:1434)
E/AndroidRuntime(28059): 	at android.view.View.layout(View.java:14824)
E/AndroidRuntime(28059): 	at android.view.ViewGroup.layout(ViewGroup.java:4631)
E/AndroidRuntime(28059): 	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:453)
E/AndroidRuntime(28059): 	at android.widget.FrameLayout.onLayout(FrameLayout.java:388)
E/AndroidRuntime(28059): 	at android.view.View.layout(View.java:14824)
E/AndroidRuntime(28059): 	at android.view.ViewGroup.layout(ViewGroup.java:4631)
E/AndroidRuntime(28059): 	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:453)
E/AndroidRuntime(28059): 	at android.widget.FrameLayout.onLayout(FrameLayout.java:388)
E/AndroidRuntime(28059): 	at android.view.View.layout(View.java:14824)
E/AndroidRuntime(28059): 	at android.view.ViewGroup.layout(ViewGroup.java:4631)
E/AndroidRuntime(28059): 	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:453)
E/AndroidRuntime(28059): 	at android.widget.FrameLayout.onLayout(FrameLayout.java:388)
E/AndroidRuntime(28059): 	at android.view.View.layout(View.java:14824)
E/AndroidRuntime(28059): 	at android.view.ViewGroup.layout(ViewGroup.java:4631)
E/AndroidRuntime(28059): 	at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1671)
E/AndroidRuntime(28059): 	at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1525)
E/AndroidRuntime(28059): 	at android.widget.LinearLayout.onLayout(LinearLayout.java:1434)
E/AndroidRuntime(28059): 	at android.view.View.layout(View.java:14824)
E/AndroidRuntime(28059): 	at android.view.ViewGroup.layout(ViewGroup.java:4631)
E/AndroidRuntime(28059): 	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:453)
E/AndroidRuntime(28059): 	at android.widget.FrameLayout.onLayout(FrameLayout.java:388)
E/AndroidRuntime(28059): 	at android.view.View.layout(View.java:14824)
E/AndroidRuntime(28059): 	at android.view.ViewGroup.layout(ViewGroup.java:4631)
E/AndroidRuntime(28059): 	at android.view.ViewRootImpl.performLayout(ViewRootImpl.java:1983)
E/AndroidRuntime(28059): 	at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:1740)
E/AndroidRuntime(28059): 	at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:996)
E/AndroidRuntime(28059): 	at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:5600)
E/AndroidRuntime(28059): 	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:761)
E/AndroidRuntime(28059): 	at android.view.Choreographer.doCallbacks(Choreographer.java:574)
E/AndroidRuntime(28059): 	at android.view.Choreographer.doFrame(Choreographer.java:544)
E/AndroidRuntime(28059): 	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:747)
E/AndroidRuntime(28059): 	at android.os.Handler.handleCallback(Handler.java:733)
E/AndroidRuntime(28059): 	at android.os.Handler.dispatchMessage(Handler.java:95)
E/AndroidRuntime(28059): 	at android.os.Looper.loop(Looper.java:136)
E/AndroidRuntime(28059): 	at android.app.ActivityThread.main(ActivityThread.java:5001)
E/AndroidRuntime(28059): 	at java.lang.reflect.Method.invokeNative(Native Method)
E/AndroidRuntime(28059): 	at java.lang.reflect.Method.invoke(Method.java:515)
E/AndroidRuntime(28059): 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:785)
E/AndroidRuntime(28059): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:601)
E/AndroidRuntime(28059): 	at dalvik.system.NativeStart.main(Native Method)
W/ActivityManager(24365):   Force finishing activity com.xabber.androiddev/com.xabber.android.ui.ContactList
```

Also when I typed something to a contact, then pressed back button, Xabber crashed, here's the logcat:

```
I/com.xabber.android.ui.ChatViewerFragment$5@422a0a68(28313): afterTextChanged
I/com.xabber.android.ui.ChatViewerFragment$5@422a0a68(28313): afterTextChanged
I/com.xabber.android.ui.ChatViewerFragment$5@422a0a68(28313): afterTextChanged
I/System.out(28313): 07:43:38 PM SENT (1108544816): <message id="uWDyB-27" to="friend@server.net" type="chat"><body>text</body><thread>s23vfhvIcf2y</thread><active xmlns="http://jabber.org/protocol/chatstates" /><request xmlns="urn:xmpp:receipts" /></message>
I/System.out(28313): 07:43:39 PM RCV  (1108544816): <message from='friend@server.net/androidtivfHLG5' to='myself@jabber.org/android2uV73Hcp' id='h98Y9-63'><thread>s23vfhvIcf2y</thread><received id='uWDyB-27' xmlns='urn:xmpp:receipts'/></message>
I/com.xabber.android.ui.ChatViewer@4208d208(28313): onPause
I/com.xabber.android.ui.ContactList@41ca4808(28313): onResume
D/AndroidRuntime(28313): Shutting down VM
W/dalvikvm(28313): threadid=1: thread exiting with uncaught exception (group=0x4184dba8)
E/AndroidRuntime(28313): FATAL EXCEPTION: main
E/AndroidRuntime(28313): Process: com.xabber.androiddev, PID: 28313
E/AndroidRuntime(28313): java.lang.NullPointerException
E/AndroidRuntime(28313): 	at com.xabber.android.ui.adapter.GroupedContactAdapter.getView(GroupedContactAdapter.java:182)
E/AndroidRuntime(28313): 	at android.widget.AbsListView.obtainView(AbsListView.java:2257)
E/AndroidRuntime(28313): 	at android.widget.ListView.makeAndAddView(ListView.java:1790)
E/AndroidRuntime(28313): 	at android.widget.ListView.fillSpecific(ListView.java:1337)
E/AndroidRuntime(28313): 	at android.widget.ListView.layoutChildren(ListView.java:1608)
E/AndroidRuntime(28313): 	at android.widget.AbsListView.onLayout(AbsListView.java:2089)
E/AndroidRuntime(28313): 	at android.view.View.layout(View.java:14824)
E/AndroidRuntime(28313): 	at android.view.ViewGroup.layout(ViewGroup.java:4631)
E/AndroidRuntime(28313): 	at android.widget.RelativeLayout.onLayout(RelativeLayout.java:1055)
E/AndroidRuntime(28313): 	at android.view.View.layout(View.java:14824)
E/AndroidRuntime(28313): 	at android.view.ViewGroup.layout(ViewGroup.java:4631)
E/AndroidRuntime(28313): 	at android.widget.RelativeLayout.onLayout(RelativeLayout.java:1055)
E/AndroidRuntime(28313): 	at android.view.View.layout(View.java:14824)
E/AndroidRuntime(28313): 	at android.view.ViewGroup.layout(ViewGroup.java:4631)
E/AndroidRuntime(28313): 	at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1671)
E/AndroidRuntime(28313): 	at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1525)
E/AndroidRuntime(28313): 	at android.widget.LinearLayout.onLayout(LinearLayout.java:1434)
E/AndroidRuntime(28313): 	at android.view.View.layout(View.java:14824)
E/AndroidRuntime(28313): 	at android.view.ViewGroup.layout(ViewGroup.java:4631)
E/AndroidRuntime(28313): 	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:453)
E/AndroidRuntime(28313): 	at android.widget.FrameLayout.onLayout(FrameLayout.java:388)
E/AndroidRuntime(28313): 	at android.view.View.layout(View.java:14824)
E/AndroidRuntime(28313): 	at android.view.ViewGroup.layout(ViewGroup.java:4631)
E/AndroidRuntime(28313): 	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:453)
E/AndroidRuntime(28313): 	at android.widget.FrameLayout.onLayout(FrameLayout.java:388)
E/AndroidRuntime(28313): 	at android.view.View.layout(View.java:14824)
E/AndroidRuntime(28313): 	at android.view.ViewGroup.layout(ViewGroup.java:4631)
E/AndroidRuntime(28313): 	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:453)
E/AndroidRuntime(28313): 	at android.widget.FrameLayout.onLayout(FrameLayout.java:388)
E/AndroidRuntime(28313): 	at android.view.View.layout(View.java:14824)
E/AndroidRuntime(28313): 	at android.view.ViewGroup.layout(ViewGroup.java:4631)
E/AndroidRuntime(28313): 	at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1671)
E/AndroidRuntime(28313): 	at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1525)
E/AndroidRuntime(28313): 	at android.widget.LinearLayout.onLayout(LinearLayout.java:1434)
E/AndroidRuntime(28313): 	at android.view.View.layout(View.java:14824)
E/AndroidRuntime(28313): 	at android.view.ViewGroup.layout(ViewGroup.java:4631)
E/AndroidRuntime(28313): 	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:453)
E/AndroidRuntime(28313): 	at android.widget.FrameLayout.onLayout(FrameLayout.java:388)
E/AndroidRuntime(28313): 	at android.view.View.layout(View.java:14824)
E/AndroidRuntime(28313): 	at android.view.ViewGroup.layout(ViewGroup.java:4631)
E/AndroidRuntime(28313): 	at android.view.ViewRootImpl.performLayout(ViewRootImpl.java:1983)
E/AndroidRuntime(28313): 	at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:1740)
E/AndroidRuntime(28313): 	at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:996)
E/AndroidRuntime(28313): 	at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:5600)
E/AndroidRuntime(28313): 	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:761)
E/AndroidRuntime(28313): 	at android.view.Choreographer.doCallbacks(Choreographer.java:574)
E/AndroidRuntime(28313): 	at android.view.Choreographer.doFrame(Choreographer.java:544)
E/AndroidRuntime(28313): 	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:747)
E/AndroidRuntime(28313): 	at android.os.Handler.handleCallback(Handler.java:733)
E/AndroidRuntime(28313): 	at android.os.Handler.dispatchMessage(Handler.java:95)
E/AndroidRuntime(28313): 	at android.os.Looper.loop(Looper.java:136)
E/AndroidRuntime(28313): 	at android.app.ActivityThread.main(ActivityThread.java:5001)
E/AndroidRuntime(28313): 	at java.lang.reflect.Method.invokeNative(Native Method)
E/AndroidRuntime(28313): 	at java.lang.reflect.Method.invoke(Method.java:515)
E/AndroidRuntime(28313): 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:785)
E/AndroidRuntime(28313): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:601)
E/AndroidRuntime(28313): 	at dalvik.system.NativeStart.main(Native Method)
W/ActivityManager(24365):   Force finishing activity com.xabber.androiddev/com.xabber.android.ui.ContactList
I/WindowManager(24365): Screenshot max retries 4 of Token{4215ca30 ActivityRecord{421d0dd0 u0 com.xabber.androiddev/com.xabber.android.ui.ContactList t16 f}} appWin=Window{42104f98 u0 com.xabber.androiddev/com.xabber.android.ui.ContactList} drawState=1
W/WindowManager(24365): Screenshot failure taking screenshot for (720x1280) to layer 21010
W/ActivityManager(24365): Activity pause timeout for ActivityRecord{421d0dd0 u0 com.xabber.androiddev/com.xabber.android.ui.ContactList t16 f}
```

Both point to: com.xabber.android.ui.adapter.GroupedContactAdapter.getView(GroupedContactAdapter.java:182)

**Comment by grigoryfedorov** [comment]
Thank you. Will be corrected tomorrow.

**Comment by grigoryfedorov** [comment]
Fixed in 30fbef38d84e79bfefd8b95dcd20ff1c33b02d68

**Comment by aelmahmoudy** [comment]
Thanks
